promgen/static/js/promgen.js

```diff
@@ -138,3 +138,22 @@ $(document).ready(function() {
     }
   });
 });
+
+// Activate a tab based on the hash value of the URL
+// For example, if the URL is `http://example.com#tab2`, it will activate the tab with id `tab2`.
+$(document).ready(function () {
+  function activateTabFromHash() {
+    var hash = window.location.hash;
+    if (hash && $(hash).length) {
+      // Remove active class from all tabs and tab content
+      $(".nav-tabs li").removeClass("active");
+      $(".tab-pane").removeClass("active");
+
+      // Add active class to tab and tab content that matches the hash
+      $('a[href="' + hash + '"]').parent().addClass("active");
+      $(hash).addClass("active");
+    }
+  }
+
+  activateTabFromHash();
+});
```

Comment:
On line number `111`, there already is a `$(document).ready` function.
Please, define your `activateTabFromHash` function outside, and call it from inside.